DDDEastAnglia.Tests/Helpers/LoginMethods/LoginMethodIconProviderTests.cs

```diff
@@ -0,0 +1,35 @@
+﻿using DDDEastAnglia.Helpers.AppSettings;
+using DDDEastAnglia.Helpers.LoginMethods;
+using NSubstitute;
+using NUnit.Framework;
+
+namespace DDDEastAnglia.Tests.Helpers.LoginMethods
+{
+    [TestFixture]
+    public sealed class LoginMethodIconProviderTests
+    {
+        [Test]
+        public void GetIcon_ReturnsTheConfiguredIconForOAuthProvider()
+        {
+            var appSettingsProvider = Substitute.For<IAppSettingsProvider>();
+            appSettingsProvider.GetSetting(LoginMethodIconProvider.AppSettingsKeyPrefix + "test").Returns("test-icon");
```

Comment:
I tend to agree (although appSettings are really just pairs of keys and values), but in this case I think it buys us something. The [original version of the icon provider](https://github.com/adrianbanks/dddeastanglia/blob/de683c0e21dfbfaf8f30210a7b27370a72d6e9af/DDDEastAnglia/Helpers/LoginMethods/LoginMethodIconProvider.cs) has bootstrap icon names embedded in the C#, which didn't feel right. Doing it this way means that we can change them without the need to recompile.